fmt/src/selection.rs

```diff
@@ -182,15 +180,45 @@ fn select_files_with_git(
         builder.build().context(SelectWithIgnoreSnafu)?
     };
 
-    let mut it = WalkDir::new(basedir).into_iter();
+    let basedir = basedir
+        .canonicalize()
+        .with_context(|_| ResolveAbsolutePathSnafu {
+            path: basedir.display().to_string(),
+        })?;
+    let mut it = WalkDir::new(basedir).follow_links(false).into_iter();
```

Comment:
`follow_links(false)` is the default, and I make it explicit as it's important to properly identify symlinks as symlinks, otherwise these would show up as the their target type. For excludes, `git` will treat them as symlinks though, so `is_dir` has to be false for symlinks.

Reply:
> make it explicit

This is correct.

I skip the select symlinks because if the target is outside the basedir, it's irrelevant; if it's inside, the real content is checked/formatted at the target.